main.go

```diff
@@ -179,13 +176,29 @@ func main() {
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		runGRPCServer(ctx, dsn, grpcBind)
+		runGRPCServer(ctx, db, *grpcBind)
 	}()
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		runJSONServer(ctx, grpcBind, jsonBind)
+		runJSONServer(ctx, *grpcBind, *jsonBind, *swaggerBind)
 	}()
+
+	ticker := time.NewTicker(24 * time.Hour)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		// Drop old partitions when api just started.
+		DropOldPartition(db, *dataRetention)
+		for {
+			select {
+			case <-ticker.C:
+				// Drop old partitions once in 24h.
+				DropOldPartition(db, *dataRetention)
+			}
```

Comment:
let's add 
```
case <-ctx.Done():
return
```

Reply:
Without it, there is an infinite loop, and qan-api can't shutdown properly. That is a serious enough issue to request changes.